### PR TITLE
Publish to Windows Package Managar (WinGet)

### LINKED ISSUE #449: [New Package]: Suwayomi.Tachidesk-Server
### List the PackageIdentifier of the package to be added to automation

Suwayomi.Tachidesk-Server

### Details about the application

https://github.com/Suwayomi/Tachidesk-Server

### PR DESCRIPTION
**Pull request by vedantmgoyal9**
This action automatically generates manifests for WinGet Community Repository ([microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs)) and submits them.

Before merging this:
1. Please add a GitHub token with `public_repo` scope as a repository secret and rename the secret name in the workflow.
2. Fork [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs) under @Suwayomi.

* Resolves https://github.com/vedantmgoyal2009/vedantmgoyal2009/issues/449

### CODE REVIEW

**Comment by vedantmgoyal9** [comment]
cc @syer10

**Comment by Syer10** [comment]
Same thing I said in the JUI Repo, just rename it to WINGET_PUBLISH_PAT, but I will also leave @AriaMoradi to do the token and merge since he is the Server project lead

**Review by Syer10 — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
       - name: Upload Release
         uses: xresloader/upload-to-github-release@v1
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.WINGET_PUBLISH_PAT }}
```
Comment (AriaMoradi) [inline .github/workflows/publish.yml:75]: why would you change this token? doesn't the other workflow publish to winget?

File: .github/workflows/winget.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  release:
+    types: [released]
+jobs:
+  publish:
+    runs-on: windows-latest # action can only be run on windows
+    steps:
+      - uses: vedantmgoyal2009/winget-releaser@latest
+        with:
+          identifier: Suwayomi.Tachidesk-Server
+          token: ${{ secrets.WINGET_PUBLISH_PAT }}
```
Comment (AriaMoradi) [inline .github/workflows/winget.yml:12]: shouldn't it be `WINGET_PUBLISH_PATH`? which perms does the token need?

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: .github/workflows/winget.yml (hunk 2 above)
Comment (vedantmgoyal9) [reply .github/workflows/winget.yml:12]: `public_repo`

File: .github/workflows/publish.yml (hunk 1 above)
Comment (vedantmgoyal9) [reply .github/workflows/publish.yml:75]: > When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.

If the release author is github-actions, it won't trigger any workflow runs and hence the publish-to-winget won't be triggered.

File: .github/workflows/winget.yml (hunk 2 above)
Comment (Syer10) [reply .github/workflows/winget.yml:12]: The PAT stands for Personal Access Token

Comment (vedantmgoyal9) [reply .github/workflows/winget.yml:12]: Yes

**Pull request merged**